### DISCUSSION TITLE
Can't install version 1.0.1 via npm

### DISCUSSION THREAD

**Discussion by aphor1st** [samizdatco/skia-canvas#216] [Q&A]
npm ERR! code 101
npm ERR! path C:\Users\zs\Desktop\projet protect\node_modules\skia-canvas
npm ERR! command failed
npm ERR! command C:\Windows\system32\cmd.exe /d /s /c node-pre-gyp install || npm run build -- --release
npm ERR! response status 403 Forbidden on https://skia-canvas.s3.us-east-1.amazonaws.com/v1.0.1/win32-ia32-napi-v6-unknown.tar.gz
npm ERR!
npm ERR! > skia-canvas@1.0.1 build
npm ERR! > cargo-cp-artifact -nc lib/v6/index.node -- cargo build --message-format=json-render-diagnostics "--release"
npm ERR! node-pre-gyp info it worked if it ends with ok
npm ERR! node-pre-gyp info using node-pre-gyp@1.0.10
npm ERR! node-pre-gyp info using node@16.13.1 | win32 | ia32
npm ERR! node-pre-gyp info check checked for "C:\Users\zs\Desktop\projet protect\node_modules\skia-canvas\lib\v6\index.node" (not found)
npm ERR! node-pre-gyp http GET https://skia-canvas.s3.us-east-1.amazonaws.com/v1.0.1/win32-ia32-napi-v6-unknown.tar.gz
npm ERR! node-pre-gyp ERR! install response status 403 Forbidden on https://skia-canvas.s3.us-east-1.amazonaws.com/v1.0.1/win32-ia32-napi-v6-unknown.tar.gz
npm ERR! node-pre-gyp ERR! install error
npm ERR! node-pre-gyp ERR! stack Error: response status 403 Forbidden on https://skia-canvas.s3.us-east-1.amazonaws.com/v1.0.1/win32-ia32-napi-v6-unknown.tar.gz
npm ERR! node-pre-gyp ERR! stack     at C:\Users\zs\Desktop\projet protect\node_modules\@mapbox\node-pre-gyp\lib\install.js:67:15
npm ERR! node-pre-gyp ERR! stack     at processTicksAndRejections (node:internal/process/task_queues:96:5)
npm ERR! node-pre-gyp ERR! System Windows_NT 10.0.19045
npm ERR! node-pre-gyp ERR! command "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Users\\zs\\Desktop\\projet protect\\node_modules\\@mapbox\\node-pre-gyp\\bin\\node-pre-gyp" "install"
npm ERR! node-pre-gyp ERR! cwd C:\Users\zs\Desktop\projet protect\node_modules\skia-canvas
npm ERR! node-pre-gyp ERR! node -v v16.13.1
npm ERR! node-pre-gyp ERR! node-pre-gyp -v v1.0.10
npm ERR! node-pre-gyp ERR! not ok
npm ERR! error: could not find `Cargo.toml` in `C:\Users\zs\Desktop\projet protect\node_modules\skia-canvas` or any parent directory
npm ERR! Did not copy "cdylib:skia-canvas"

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\zs\AppData\Local\npm-cache\_logs\2022-12-03T21_31_11_459Z-debug.log

C:\Users\zs\Desktop\projet protect>npm i skia-canvas
npm ERR! code 101
npm ERR! path C:\Users\zs\Desktop\projet protect\node_modules\skia-canvas
npm ERR! command failed
npm ERR! command C:\Windows\system32\cmd.exe /d /s /c node-pre-gyp install || npm run build -- --release
npm ERR! response status 403 Forbidden on https://skia-canvas.s3.us-east-1.amazonaws.com/v1.0.1/win32-ia32-napi-v6-unknown.tar.gz
npm ERR!
npm ERR! > skia-canvas@1.0.1 build
npm ERR! > cargo-cp-artifact -nc lib/v6/index.node -- cargo build --message-format=json-render-diagnostics "--release"
npm ERR! node-pre-gyp info it worked if it ends with ok
npm ERR! node-pre-gyp info using node-pre-gyp@1.0.10
npm ERR! node-pre-gyp info using node@16.13.1 | win32 | ia32
npm ERR! node-pre-gyp info check checked for "C:\Users\zs\Desktop\projet protect\node_modules\skia-canvas\lib\v6\index.node" (not found)
npm ERR! node-pre-gyp http GET https://skia-canvas.s3.us-east-1.amazonaws.com/v1.0.1/win32-ia32-napi-v6-unknown.tar.gz
npm ERR! node-pre-gyp ERR! install response status 403 Forbidden on https://skia-canvas.s3.us-east-1.amazonaws.com/v1.0.1/win32-ia32-napi-v6-unknown.tar.gz
npm ERR! node-pre-gyp ERR! install error
npm ERR! node-pre-gyp ERR! stack Error: response status 403 Forbidden on https://skia-canvas.s3.us-east-1.amazonaws.com/v1.0.1/win32-ia32-napi-v6-unknown.tar.gz
npm ERR! node-pre-gyp ERR! stack     at C:\Users\zs\Desktop\projet protect\node_modules\@mapbox\node-pre-gyp\lib\install.js:67:15
npm ERR! node-pre-gyp ERR! stack     at processTicksAndRejections (node:internal/process/task_queues:96:5)
npm ERR! node-pre-gyp ERR! System Windows_NT 10.0.19045
npm ERR! node-pre-gyp ERR! command "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Users\\zs\\Desktop\\projet protect\\node_modules\\@mapbox\\node-pre-gyp\\bin\\node-pre-gyp" "install"
npm ERR! node-pre-gyp ERR! cwd C:\Users\zs\Desktop\projet protect\node_modules\skia-canvas
npm ERR! node-pre-gyp ERR! node -v v16.13.1
npm ERR! node-pre-gyp ERR! node-pre-gyp -v v1.0.10
npm ERR! node-pre-gyp ERR! not ok
npm ERR! error: could not find `Cargo.toml` in `C:\Users\zs\Desktop\projet protect\node_modules\skia-canvas` or any parent directory
npm ERR! Did not copy "cdylib:skia-canvas"

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\zs\AppData\Local\npm-cache\_logs\2022-12-03T21_32_11_360Z-debug.log

**Comment by krushnaBpawar** [comment]
I am also facing same error

**Comment by MauriceNorden** [comment]
It seems like the package want's to build the binary put can't download the dependencies since doing that gives out a 403 error wich means "Forbidden".
You can try to download the repo, install all the build dependecies as descibed in the README.md, and run `npm run build` in the root folder and move the files to `node_modules/skia-canvas`

**Comment by kostia1st** [comment]
```
403 Forbidden on https://skia-canvas.s3.us-east-1.amazonaws.com/v1.0.1/linux-x64-napi-v6-glibc.tar.gz
```

Just today. Seems like a recurring issue

**Comment by ghost** [comment]
@kostia1st the solution is mentioned here, https://github.com/samizdatco/skia-canvas/issues/160
